apps/prairielearn/src/pages/instructorAssessmentQuestions/components/InstructorAssessmentQuestionsTable.tsx

```diff
@@ -47,81 +58,915 @@ function Title({
+  max_auto_points,
+  max_manual_points,
+  auto_points,
+  assessmentType,
+}: {
+  max_auto_points: number | number[] | null;
+  max_manual_points: number | null;
+  auto_points: number | number[] | null;
+  assessmentType: EnumAssessmentType;
+}) {
+  if (auto_points || !max_manual_points) {
+    if (assessmentType === 'Exam') {
+      const pointsArray = Array.isArray(auto_points)
+        ? auto_points
+        : [auto_points ?? max_manual_points];
+      return pointsArray.map((p) => (p ?? 0) - (max_manual_points ?? 0)).join(',');
+    }
+    if (assessmentType === 'Homework') {
+      const initPointsValue = Array.isArray(auto_points) ? auto_points[0] : auto_points;
+      const maxAutoPointsValue = Array.isArray(max_auto_points)
+        ? max_auto_points[0]
+        : max_auto_points;
+      return `${(initPointsValue ?? 0) - (max_manual_points ?? 0)}/${maxAutoPointsValue ?? 0}`;
+    }
+  } else {
+    return '—';
+  }
+}
+
+function mapQuestions(
+  course: StaffCourse,
+  rows: StaffAssessmentQuestionRow[],
+): ZoneAssessmentJson[] {
+  const zones: ZoneAssessmentJson[] = [];
+  const zoneAlternativeGroupCounts: Record<number, number> = {};
+
+  for (const row of rows) {
+    if (row.zone.number == null) throw new Error('Zone number required');
+
+    zones[row.zone.number - 1] ??= {
+      title: row.zone.title ?? undefined,
+      comment: row.zone.json_comment ?? undefined,
+      maxPoints: row.zone.max_points ?? undefined,
+      numberChoose: row.zone.number_choose ?? undefined,
+      bestQuestions: row.zone.best_questions ?? undefined,
+      questions: [],
+      advanceScorePerc: row.zone.advance_score_perc ?? undefined,
+      gradeRateMinutes: row.zone.json_grade_rate_minutes ?? undefined,
+      canView: row.zone.json_can_view ?? [],
+      canSubmit: row.zone.json_can_submit ?? [],
+    };
+
+    if (row.alternative_group.number == null) throw new Error('Alternative group number required');
+
+    const zoneNumber = row.zone.number;
+    zoneAlternativeGroupCounts[zoneNumber] ??= -1;
+
+    // If this is a new alternative group in this zone, increment the count
+    if (row.start_new_alternative_group) {
+      zoneAlternativeGroupCounts[zoneNumber]++;
+    }
+
+    // Use the count as the position within the zone
+    const positionInZone = zoneAlternativeGroupCounts[zoneNumber];
+    if (!zones[zoneNumber - 1].questions[positionInZone]) {
+      zones[zoneNumber - 1].questions[positionInZone] ??= {
+        id: row.alternative_group.id ? questionDisplayName(course, row) : undefined,
+        comment: row.alternative_group.json_comment ?? undefined,
+        advanceScorePerc: row.alternative_group.advance_score_perc ?? undefined,
+        canView: row.alternative_group.json_can_view ?? [],
+        canSubmit: row.alternative_group.json_can_submit ?? [],
+        gradeRateMinutes: row.alternative_group.json_grade_rate_minutes ?? undefined,
+        numberChoose: row.alternative_group.number_choose ?? 1,
```

Comment:
Is this correct? Consider the following zone:

```json
    {
      "questions": [
        {
          "alternatives": [
            { "id": "partialCredit1", "advanceScorePerc": 0 },
            { "id": "partialCredit2", "advanceScorePerc": 50 }
          ],
          "points": 1
        }
      ]
    },
```

When rendered in the UI, this shows "Choose 1 question from" in the alternative group. You should ensure that this syncs correctly too.